README.md

```diff
@@ -1,103 +1,122 @@
-```
+* Value type is [string](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#string)
+* Required
 
-At this point any modifications to the plugin code will be applied to this local Logstash setup. After modifying the plugin, simply rerun Logstash.
+This is the full URL of the [Generic log ingest API v2](https://www.dynatrace.com/support/help/how-to-use-dynatrace/log-monitoring/log-monitoring-v2/post-log-ingest/) endpoint on your ActiveGate.
+Example: `"ingest_endpoint_url" => "https://abc123456.live.dynatrace.com/api/v2/logs/ingest"`
 
-#### 2.2 Run in an installed Logstash
+### `api_key`
 
-You can use the same **2.1** method to run your plugin in an installed Logstash by editing its `Gemfile` and pointing the `:path` to your local plugin development directory or you can build the gem and install it using:
+* Value type is [string](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#string)
+* Required
 
-- Build your plugin gem
+This is the [Dynatrace API token](https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/) which will be used to authenticate log ingest requests.
+It requires the `logs.ingest` (Ingest Logs) scope to be set and it is recommended to limit scope to only this one.
+Example: `"api_key" => "dt0c01.4XLO3..."`
 
-```sh
-gem build logstash-output-dynatrace.gemspec
+### `ssl_verify_none`
 
-```
+* Value type is [boolean](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#boolean)
+* Optional
+* Default value is `false`
 
-- Install the plugin from the Logstash home
+It is recommended to leave this optional configuration set to `false` unless absolutely required.
+Setting `ssl_verify_none` to `true` causes the output plugin to skip certificate verification when sending log ingest requests to SSL and TLS protected HTTPS endpoints.
+This option may be required if you are using a self-signed certificate, an expired certificate, or a certificate which was generated for a different domain than the one in use.
 
-```sh
-# Logstash 2.3 and higher
-bin/logstash-plugin install --no-verify
+### `codec` 
 
-# Prior to Logstash 2.3
-bin/plugin install --no-verify
-```
+* Value type is codec
+* Default value is "plain"
+* 
+The codec used for output data. Output codecs are a convenient method for encoding your data before it leaves the output without needing a separate filter in your Logstash pipeline.
 
-- Start Logstash and proceed to test the plugin
+### `enable_metric`
 
-## Contributing
+* Value type is boolean
+* Default value is true
 
-All contributions are welcome: ideas, patches, documentation, bug reports, complaints, and even something you drew up on a napkin.
+Disable or enable metric logging for this specific plugin instance. By default we record all the metrics we can, but you can disable metrics collection for a specific plugin.
```

Comment:
Does this also send metrics to Dynatrace?

Reply:
It generates metrics with a specific tag. You can use that tag to send metrics to Dynatrace if you want.

Reply:
Actually it just adds them to the logstash stats endpoint `/_node/stats` which exports basic statistics about the particular logstash node. Setting `enable_metric` for a particular plugin causes metrics specific to that plugin to be included in those stats. By default it is just event count.

Reply:
https://www.elastic.co/guide/en/logstash/current/node-stats-api.html

Reply:
Nice, thanks!